adapters/andalucia.js

```diff
@@ -0,0 +1,925 @@
+}
+//
+const generateTasks = (url, task, now) => {
+  return (done) => {
+    // Try getting data from today and yesterday
+      // in both cases, if unsuccessful, return blank list
+      // when successful, return list of parsed data.
+    async.parallel([
+      (cb) => {
+        request.get(
+          url, (err, res, body) => {
+            if (err || res.statusCode !== 200) {
+              console.log(
+                'Records for ' + moment().format('DD-MM-YY') +
+                ' not reached.'
+              );
+              return cb(null, []);
+            }
+            let $ = cheerio.load(body);
+            const data = formatData($, stations);
+            return cb(null, data);
+          }
+        );
+      },
+      (cb) => {
+        url = url.replace(now.format('YYMMDD'), moment().add(-1, 'days').format('YYMMDD'));
+        request.get(
+          url, (err, res, body) => {
+            if (err || res.statusCode !== 200) {
+              console.log(
+                'Records for ' + moment().add(-1, 'days').format('DD-MM-YY') +
+                'not reached.'
+              );
+              return cb(null, []);
+            }
+            let $ = cheerio.load(body);
+            const data = formatData($, stations);
+            cb(null, data);
+          }
+        );
+      }], (err, res) => {
+      if (err) {
+        return done(null, []);
+      }
+      res = res[0].concat(res[1]);
+      done(null, res);
+    });
+  };
+};
+
+const formatData = ($, stations) => {
+  const tables = [];
+  // the tables we want don't have cellpadding 5.
+  $('table').each((i, el) => {
+    const table = $(el)['0'];
+    if (table.attribs.cellpadding !== '5') {
+      tables.push($(el));
+    }
+  });
+  // data is held two consecutive tables
+  // (one with station name, other actual data)
+  let stationData = _.chunk(tables, 2);
+  // return a list of measurement objects for each table.
+  return stationData.map((table) => {
+    const station = $(table[0]['0'].children[2]).text().split('Estacion')[1].trim();
+    const stationLoc = mapStationCoords(station);
+    table = makeTable(table[1]['0'], $);
+    return makeMeasurements(table, stationLoc);
+  });
+};
+
+const mapStationCoords = (station) => {
+  return stations.filter((stationObj) => {
+    return stationObj['name'] === station;
+  }).map((stationObj) => {
+    return {
+      longitude: stationObj['coordinates'][0],
+      latitude: stationObj['coordinates'][1]
+    };
+  });
+};
+
+const makeTable = (cheerioTable, $) => {
+  return cheerioTable.children.filter((child) => {
+    return child.name === 'tr';
+  }).map((row) => {
+    const tds = [];
+    $(row).children().each((i, el) => {
+      tds.push($(el).text());
+    });
+    return tds;
+  });
+};
+
+const makeMeasurements = (stationData, stationLoc) => {
+  // get index of data sources we can record
+  const pollutantIndexes = stationData[0].filter((cell) => {
+    return _.includes(['SO2', 'NO2'], cell);
```

Comment:
Can you use the utils function acceptableParameters here to make it less hard coded?